src/backend/EinsumNode.cpp

```diff
@@ -400,14 +439,20 @@ einsum_ir::err_t einsum_ir::backend::EinsumNode::unlock_data() {
     return err_t::NO_DATA_PTR_PROVIDED;
   }
 
+  delete [] (char *) m_data_ptr_int;
```

Comment:
This is not clear to me: Shouldn't the free happen inside of the memory manager?

Reply:
This free is for the locked data, which is saved in a separate array. It does not influence the memory manager. 
I will change the code structure to make it clearer.